boards/shared/Src/Io/Io_Adc.c

```diff
@@ -0,0 +1,22 @@
+#include "Io_Adc.h"
+
+extern uint16_t adc_readings[NUM_ADC_CHANNELS];
+extern float    v1, v2, v3, v4;
+
+void HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef *hadc)
+{
+    // If you were to over-sample this is where you can perform:
+    // Filter_ADC_Readings(ADC_CHANNEL_COUNT, ADC_FILTER_SIZE,
+    // Unfiltered_ADC_Readings, Filtered_ADC_Readings);
+
+    v1 = Convert_ADC_Voltage(adc_readings[CHANNEL_2]);
+    v2 = Convert_ADC_Voltage(adc_readings[CHANNEL_7]);
+    v3 = Convert_ADC_Voltage(adc_readings[CHANNEL_8]);
+    v4 = Convert_ADC_Voltage(adc_readings[CHANNEL_9]);
```

Comment:
For each voltage/current measurement you would want to implement a function in _IoAdc.c_ that would look something like:
```
float Get_Vbat(uint16_t adc_value,)
{
	float voltage = Convert_ADC_Voltage(adc_value);
	return voltage*VBAT_VOLTAGE_GAIN_RATIO;
}

```